examples/example2.cpp

```diff
@@ -138,16 +93,65 @@ simdjson_inline simdjson_result<T> simdjson::ondemand::value::get() noexcept {
   return container;
 }
 
-// Specialization for std::vector<int>
-template <>
-simdjson_inline simdjson_result<std::vector<int> > simdjson::ondemand::value::get() noexcept {
-  return get<std::vector<int>>();
-}
+template <typename T>
+simdjson_inline simdjson_result<T>
+simdjson::ondemand::document::get() & noexcept {
+  if constexpr (PushableContainer<T>) {
+    ondemand::array array;
+    auto error = get_array().get(array);
+    if (error) {
+      return error;
+    }
+    T container;
+    for (auto v : array) {
+      typename T::value_type val;
+      if constexpr (std::is_same_v<typename T::value_type, double>) {
```

Comment:
Hi, sorry for the intrusion, just wanted to tell that you can make this a bit cleaner with consteval lambda like following: https://godbolt.org/z/16M9h8Wbd
```
auto type_is = []<typename compare>(compare v) consteval {
        return std::same_as<typename T::value_type, compare>;
};
if constexpr( type_is(double{}) ) {
        std::println("double");
} else {
        std::println("not_double");
}
```

Reply:
Thanks. That's reasonable.